src/common/extensionMessaging.ts

```diff
@@ -28,6 +29,17 @@ export interface IExtensionMessageSender {
     sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any>;
 }
 
+export function validateMessageWithArguments(message: MessageWithArguments): void {
```

Comment:
I just tried a small example, and the compiler does not accept `string` instead of `any[]`. Again, we should fix this at compile time, not runtime.